kernel/mmanutils.c

```diff
@@ -960,15 +960,46 @@ int myst_msync(void* addr, size_t length, int flags)
 
     _rlock(&locked);
     {
+        int prot;
+        bool consistent;
         const size_t n = length / PAGE_SIZE;
-        const uint8_t* page = addr;
+        uint8_t* page = addr;
+        char* prev_pathname = NULL;
 
         for (size_t i = index; i < index + n; i++)
         {
             myst_fdmapping_t* p = &v.fdmappings[i];
-
-            if (p->used == MYST_FDMAPPING_USED)
-                ECHECK(_sync_file(p->fd, p->offset, page, PAGE_SIZE));
+            /* sync only for writable mappings */
+            ECHECK(myst_mman_get_prot(
+                &_mman, page, PAGE_SIZE, &prot, &consistent));
```

Comment:
`consistent` is not used. Do we need a check for it as well?

Reply:
Protection is being checked for a single page here, so it should always be consistent.

Reply:
Ah, I see. In this case, you can simply pass NULL instead of `&consistent`

Reply:
`myst_mman_get_prot` does not allow a nullable `consistent` argument currently.